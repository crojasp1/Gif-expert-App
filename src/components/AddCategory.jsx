import PropTypes from "prop-types";
import { useState } from "react"


const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <= 1) return;
    //setCategories(categories => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue('');
  }

  const onInputChange = (event) => {
    setInputValue(event.target.value);

  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <div  className="input-container">

      <input 
        type="text" 
        placeholder="Buscar Gifs" 
        value={inputValue}
        onChange={onInputChange}
      ></input>

      <button className="insert-button" type="submit">Insert New Category</button>

      </div>
    </form>

    
  )
}

export default AddCategory

AddCategory.prototypes = {
  onNewCategory : PropTypes.func.isRequired,
}