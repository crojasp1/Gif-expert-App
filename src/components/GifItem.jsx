import PropTypes from "prop-types";

const GifItem = ({gifElement: {title, url}}) => {
  return (
    <div className='card'>
      <img src={url}/>
      <p>{title}</p>
    </div>
  )
}

export default GifItem;

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url  : PropTypes.string.isRequired,
}




