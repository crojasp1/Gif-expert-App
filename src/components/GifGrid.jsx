//import { getGifts } from "../helpers/getGifs";
import GifItem from "./GifItem";
import useFetchGif from "../hooks/useFetchGif";

const GifGrid = ({ category }) => {
  const { gifs, isLoading } = useFetchGif(category);

  /*
  const newStateImages = async() => {
    const images = await getGifts(category);
    setGifs(images);
  };

  useEffect(() => {
    newStateImages();
  }, []);
  */
  return (
    <>
      <h2>{category}</h2>
      {isLoading && <h3>Cargando...</h3>}
      <div className="card-grid">
        {gifs.map((element) => (
          <GifItem key={element.id} gifElement={element} />
        ))}
        ;
      </div>
    </>
  );
};

export default GifGrid;
