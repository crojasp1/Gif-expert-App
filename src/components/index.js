import GifItem from "./GifItem";
import GifGrid from "./GifGrid";
import AddCategory from "./addCategory";

export { GifItem, GifGrid, AddCategory };
