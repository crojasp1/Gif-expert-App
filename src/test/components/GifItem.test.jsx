import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components";


describe('Pruebas en <GifItem.jsx>', () => {
    const title = 'Dragon Ball';
    const url = 'https://dragon-ball.com/goku.jpg';

  test('should match with snapshot', () => {
    
    const {container} = render(<GifItem 
      gifElement={{title:title, url:url}}
      />);

      expect( container ).toMatchSnapshot();
  });

  test('should show the alt and url set', () => {
    render(<GifItem gifElement={{title:title, url:url}} />);
    //screen.debug();
    //console.log(screen.getByRole('img').alt);
    
    expect(screen.getByRole('img').src).toBe(url);
    expect(screen.getByText('Dragon Ball')).toBeTruthy();

  });
  
})
