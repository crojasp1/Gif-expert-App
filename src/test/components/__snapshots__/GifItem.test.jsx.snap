// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en <GifItem.jsx> > should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://dragon-ball.com/goku.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
